src/components/AssetsResultsCount/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedNumber } from 'react-intl';
+
+import edxBootstrap from '../../SFE.scss';
+import { hasSearchOrFilterApplied } from '../../utils/getAssetsFilters';
+import styles from './AssetsResultsCount.scss';
+import WrappedMessage from '../../utils/i18n/formattedMessageWrapper';
+import messages from './displayMessages';
+
+const renderCount = count => (
+  <span className={`${edxBootstrap['font-weight-bold']}`}>
+    <FormattedNumber value={count} />
+  </span>
+);
+
+const AssetsResultsCount = ({ paginationMetaData, filtersMetaData, searchMetaData }) => {
+  let message = messages.assetsResultsCountTotal;
+  if (hasSearchOrFilterApplied(filtersMetaData.assetTypes, searchMetaData.search)) {
+    message = messages.assetsResultsCountFiltered;
+  }
+
+  return (
+    <div className={styles['result-count-wrapper']}>
+      <WrappedMessage
+        message={message}
+        values={{
+          start: renderCount(paginationMetaData.start + 1),
+          end: renderCount(paginationMetaData.end),
+          total: renderCount(paginationMetaData.totalCount),
```

Comment:
I'm not sure if using `renderCount` like this will work correctly with the pluralization I mentioned above, I think that needs a plain number. Surely there's something for this scenario...

LMK when you get to this portion, I'll help get it ironed out if it's tricky. I'm also okay with ignoring the pluralization work in the short term (still to be done later) if it's going to delay this large PR from landing.

Reply:
Maybe [`<FormattedPlural>`](https://github.com/yahoo/react-intl/wiki/Components#formattedplural) could do it? It looks like it allows embedding the text inside elements (just like `<FormattedMessage>`). Since this means we will have to make separate displayMessages for "file"/"total files" and "possible match"/"possible matches" strings, we would need to make a `<WrappedFormattedPlural>` so each plural string is wrapped in the proper lang.

I still don't get how translators would specify other plural type translations that English doesn't need (zero, few, many, etc.) and how our code would know to use them. Do we just define a separate displayMessage for each plural type and just use the same word in English for most of the defaults?